polynote-frontend/polynote/interpreter/client_interpreter.ts

```diff
@@ -88,18 +89,23 @@ export class ClientInterpreter {
             }
 
             if (waitCellId !== undefined) {
-                return new Promise<void>(resolve => {
-                    const disposable = this.notebookState.addObserver(state => {
-                        const maybeCellReady = state.cells[waitCellId!];
-                        if (maybeCellReady && !maybeCellReady.running && !maybeCellReady.queued) {
+                let wasRunning = false;
+                return new Promise<void>((resolve, reject) => {
+                    const disposable = this.notebookState.view("cells").view(waitCellId!).addObserver(state => {
+                        if (state?.running) {
+                            wasRunning = true;
+                        }if (!state?.queued) {
```

Comment:
[formatting] can you put this `if` on a new line (initially I mistook it for an `else if`)

Reply:
Actually, is this supposed to be an `else`? Otherwise if I'm reading this right doesn't it dispose the observer when `state.running == true` ?

Reply:
Yes, I think you're right.